### PR TITLE
Yet another header cleanup

### PR DESCRIPTION
**Pull request by AngheloAlf**
- Make a `z64skin_matrix.h` header
- Make a `z64malloc.h` header
- Make a `z64lib.h` header for the `z_lib.c` functions
- Make a header for `gamealloc.h`
- Move the function declarations from `z_lights.c` to `z64lights.h` and cleanup `zlights.c` a bit
- Move "Transition" function declarations to `z64transition.h` and some cleanup on those files.
- Move `SaveContext_Init` and `gSaveContext` to `z64save.h`
- Move `D_0E000000` to `gfx.h`
- Move the `VTX` and `VTX_T` macros to `gfx.h`
- Move `SEGMENTED_TO_K0` to `main.h` because that's where `gSegments` is declared.

I was wondering if we should move `SCREEN_*`/`HIRES_BUFFER_*` defines from `macros.h` and where to put them. I was thinking to put them either on `main.h` (because of `gScreenWidth` and `gScreenHeight`) or `gfx.h`. Any ideas are welcome.

The diff is not that big, but there are many unrelated changes on this PR. I can break down this PR into multiple if that's preferred

### CODE REVIEW

File: src/code/z_fbdemo_circle.c [hunk 1]
```diff
@@ -56,8 +67,9 @@ void* TransitionCircle_Init(void* thisx) {
 void TransitionCircle_Destroy(void* thisx) {
 }
 
-void TransitionCircle_Update(void* thisx) {
+void TransitionCircle_Update(void* thisx, s32 updateRate) {
     TransitionCircle* this = (TransitionCircle*)thisx;
+    s32 unused = updateRate ? 0 : 0;
```
Comment (engineer124) [inline src/code/z_fbdemo_circle.c:72]: Huh, really. This works?

File: src/code/z_lib.c [hunk 2]
```diff
@@ -1,4 +1,12 @@
-#include "global.h"
+#include "z64lib.h"
+
+#include "main.h"
+#include "ichain.h"
+#include "sfx.h"
+#include "z64actor.h"
+#include "z64game.h"
+#include "functions.h"
+#include "macros.h" // for SQ
```
Comment (engineer124) [inline src/code/z_lib.c:9]:
```suggestion
#include "macros.h"
```
Are these comments necessary?

File: src/code/z_lights.c [hunk 3]
```diff
@@ -1,10 +1,14 @@
-#include "global.h"
+#include "z64light.h"
+
 #include "sys_cfb.h"
+#include "z64.h"
+#include "functions.h" // for SkinMatrix_Vec3fMtxFMultXYZ and Actor_GetProjectedPos
```
Comment (engineer124) [inline src/code/z_lights.c:5]: Same here (and a few other places)

Comment (AngheloAlf) [reply src/code/z_lights.c]: Sorry, those were notes for myself. I was using them to remember why those global headers were included and consider if I could make a header for them

File: src/code/z_fbdemo_circle.c (hunk 1 above)
Comment (AngheloAlf) [reply src/code/z_fbdemo_circle.c:72]: Yeah, learnt this trick on af

**Review by hensldm — APPROVED** [review]

**Review by engineer124 — APPROVED** [review]

**Review by tom-overton — APPROVED** [review]

**Pull request merged**